### PR TITLE
Remove requireJS

### PR DESCRIPTION
**Pull request by sndrs**
cc @rich-nguyen @OliverJAsh 

i _think_ this is everything, except for the following files:

```
javascripts/projects/common/modules/commercial/third-party-tags/audience-science-gateway.js
javascripts/projects/common/modules/commercial/third-party-tags/audience-science.js
javascripts/projects/common/modules/commercial/third-party-tags/criteo.js
javascripts/projects/common/modules/commercial/third-party-tags/outbrain.js
javascripts/projects/common/modules/commercial/third-party-tags/remarketing.js
```

they all need appear to need to:
- [x] handle arguments passed to the`!system-script` plugin

?

Depends on https://github.com/guardian/frontend/pull/9385.

### CODE REVIEW

**Comment by OliverJAsh** [comment]
You can remove Bower bits too!

File: static/src/systemjs-normalize.js [hunk 1]
```diff
@@ -8,13 +8,6 @@
...
     System.normalize = function (name, parentName) {
-        var requireToSystemPluginMap = {
-            // Bundled
-            'text': 'text',
-            'inlineSvg': 'svg',
-            // Runtime
-            'js': 'system-script'
-        };
         var transformers = [
             function map(name) {
```
Comment (OliverJAsh) [inline static/src/systemjs-normalize.js]: You can remove this transformer too, just update the module ID in the code.

**Comment by OliverJAsh** [comment]
We still need to turn back on the switch. I'm going to pair with @mattosborn tomorrow morning to find a fix for their interactive.

File: common/app/views/fragments/javaScriptLaterSteps.scala.html [hunk 2]
```diff
@@ -3,447 +3,314 @@
 @import conf.Static
 @import conf.Configuration
 
-@modernBrowser()
-
-@if(mvt.JspmTest.isParticipating || item.section == "crosswords") {
```
Comment (OliverJAsh) [inline common/app/views/fragments/javaScriptLaterSteps.scala.html]: Can you remove this test from the Scala codebase?

**Comment by OliverJAsh** [comment]
Nice!

**Comment by phamann** [comment]
:+1:

**Comment by OliverJAsh** [comment]
Re. your `!system-script` question, we currently have `require(['js!foo!exports=bar'])`. This is Require/curl syntax for loading a global module. You have to specify which global variable the module should export. SystemJS supports all of this out of the box—you don't need a plugin or any special syntax.

If those modules live on the same domain we can load them normally without a plugin. They will be downloaded via XHR. If they are cross domain, we will need to use `!system-script`.

**Comment by sndrs** [comment]
> You can remove Bower bits too!

bower stuff should be gone, save 2 jspm deps that still need it (for now, only, hopefully)

File: common/app/views/fragments/javaScriptLaterSteps.scala.html (hunk 2 above)
Comment (sndrs) [inline common/app/views/fragments/javaScriptLaterSteps.scala.html]: done

**Comment by OliverJAsh** [comment]
This can be merged now. SystemJS is all good to go and live for 1% of users.

**Comment by OliverJAsh** [comment]
The build is failing because we still need `static/src/javascripts/components/curl/curl-domReady.js`, it's used in other parts of the projects i.e. admin, AFAIK.

**Comment by OliverJAsh** [comment]
LGTM :+1: Nice work!

**Comment by robertberry** [comment]
Ship it :ship: :shipit: :anchor:

**Comment by OliverJAsh** [comment]
Blocker: we need to add CORS on assets-secure.guim so the interactives team
can preview their interactives. SystemJS downloads using XHR, which
requires CORS for cross-domain XHRs.

On 27 May 2015 at 11:39, Rob Berry <notifications@github.com> wrote:

> Ship it [image: :ship:] [image: :shipit:] [image: :anchor:]
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/guardian/frontend/pull/9222#issuecomment-105861630>.
>



-- 
Oliver Ash (Olly)
Software Engineer

-- 

------------------------------
Visit theguardian.com. On your mobile and tablet, download the Guardian 
iPhone and Android apps theguardian.com/guardianapp and our tablet editions 
theguardian.com/editions.  Save up to 57% by subscribing to the Guardian 
and Observer - choose the papers you want and get full digital access. 
 Visit subscribe.theguardian.com

This e-mail and all attachments are confidential and may also be 
privileged. If you are not the named recipient, please notify the sender 
and delete the e-mail and all attachments immediately. Do not disclose the 
contents to another person. You may not use the information for any 
purpose, or store, or copy, it in any way.  Guardian News & Media Limited 
is not liable for any computer viruses or other material transmitted with 
or as part of this e-mail. You should employ virus checking software.
 
Guardian News & Media Limited is a member of Guardian Media Group plc. Registered 
Office: PO Box 68164, Kings Place, 90 York Way, London, N1P 2AP.  Registered 
in England Number 908396

**Comment by robertberry** [comment]
Ok. I'm closing this until that's resolved then.

**Comment by OliverJAsh** [comment]
Have we tested the admin site?

**Comment by OliverJAsh** [comment]
Must merge https://github.com/guardian/frontend/pull/9385 before.

**Comment by sndrs** [comment]
hello again

**Comment by robertberry** [comment]
hair

HAI

**Comment by OliverJAsh** [comment]
DONE!

Merge tomorrow and then we can never speak the words "curl", "RequireJS", or "bower" again.

**Comment by phamann** [comment]
Woohooo!

**Comment by phamann** [comment]
Opps, spoke too soon. Just noticed the revert...

**Comment by OliverJAsh** [comment]
Tracking https://github.com/guardian/frontend/pull/9432.

**Pull request merged**